tests/metagpt/test_incremental_dev.py

```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+@Time    : 2024/01/03
+@Author  : mannaandpoem
+@File    : test_incremental_dev.py
+"""
+import os
+import subprocess
+
+import pytest
+from typer.testing import CliRunner
+
+from metagpt.const import DATA_PATH
+from metagpt.logs import logger
+from metagpt.startup import app
+
+runner = CliRunner()
+
+
+def test_refined_simple_calculator():
+    project_path = f"{DATA_PATH}/simple_add_calculator"
+    check_or_create_base_tag(project_path)
+
+    args = [
+        "Add subtraction, multiplication and division operations to the calculator. The current calculator can only perform basic addition operations, and it is necessary to introduce subtraction, multiplication, division operation into the calculator",
+        "--inc",
+        "--project-path",
+        project_path,
+    ]
+    result = runner.invoke(app, args)
+    logger.info(result)
+    logger.info(result.output)
+    if "Aborting" in result.output:
+        assert False
+    else:
+        tag = subprocess.run(["git", "describe", "--tags"], capture_output=True, text=True).stdout.strip()
+        # After running, there will be new commit
+        if tag == "base":
+            assert False
+        else:
+            assert True
+            try:
+                subprocess.run(["git", "tag", "refine"], check=True)
+            except subprocess.CalledProcessError as e:
+                raise e
+
+
+def test_refined_number_guessing_game():
+    project_path = f"{DATA_PATH}/number_guessing_game"
+    check_or_create_base_tag(project_path)
+
+    args = [
+        "Adding graphical interface functionality to enhance the user experience in the number-guessing game. The existing number-guessing game currently relies on command-line input for numbers. The goal is to introduce a graphical interface to improve the game's usability and visual appeal",
+        "--inc",
+        "--project-path",
+        project_path,
+    ]
+    result = runner.invoke(app, args)
+    logger.info(result)
+    logger.info(result.output)
+    if "Aborting" in result.output:
```

Comment:
Can we add a base function to only pass project_path and increment context to simplify the code. Most of them are repetitive.